01_draw_functions/01_subfunctions/ControlPlots.m

```diff
@@ -23,70 +23,89 @@ function ControlPlots(sysvector, fconv_gpsalt)
 hold on;
-plot(roll.Time, roll.Data);
-plot(rollRef.Time, rollRef.Data);
-plot(pitch.Time, pitch.Data);
-plot(pitchRef.Time, pitchRef.Data);
+plot(yaw.Time, rad2deg(yaw.Data));
+plot(yawRef.Time, rad2deg(yawRef.Data));
+hold off;
+legend('Yaw Angle', 'Yaw Angle Ref')
+%ylabel('Attitude [deg]')
+
+axeshandle(end+1) = subplot_tight(nrSubplotSections,1,[3 4],[plotmargins.vert plotmargins.horiz]);
+hold on;
+plot(roll.Time, rad2deg(roll.Data));
+plot(rollRef.Time, rad2deg(rollRef.Data));
+plot(pitch.Time, rad2deg(pitch.Data));
+plot(pitchRef.Time, rad2deg(pitchRef.Data));
 hold off;
 legend('Roll Angle', 'Roll Angle Ref', 'Pitch Angle', 'Pitch Ref');    
 ylabel('Attitude [deg]')
 
-% yaw angle plot
-control(3) = subplot(4,1,3);
+% rates plot
+axeshandle(end+1) = subplot_tight(nrSubplotSections,1,[5 6],[plotmargins.vert plotmargins.horiz]);
 hold on;
-plot(yaw.Time, yaw.Data);
-plot(yawRef.Time, yawRef.Data);
+plot(sysvector('vehicle_attitude_0.rollspeed').Time, rad2deg(sysvector('vehicle_attitude_0.rollspeed').Data));
+plot(pRef.Time, pRef.Data);
+plot(sysvector('vehicle_attitude_0.pitchspeed').Time, rad2deg(sysvector('vehicle_attitude_0.pitchspeed').Data));
+plot(qRef.Time, qRef.Data);
+plot(sysvector('vehicle_attitude_0.yawspeed').Time, rad2deg(sysvector('vehicle_attitude_0.yawspeed').Data));
+plot(rRef.Time, rRef.Data);
 hold off;
-legend('Yaw Angle', 'Yaw Angle Ref')
-ylabel('Attitude [deg]')
+legend('p','pRef','q','qRef','r','rRef');
+ylabel('Rates [deg/s]')
+
+axeshandle(end+1) = subplot_tight(nrSubplotSections,1,[7 8],[plotmargins.vert plotmargins.horiz]);
+% airspeed plots
+hold on;
+plot(sysvector('actuator_controls_0.control_0').Time, sysvector('actuator_controls_0.control_0').Data);
+plot(sysvector('actuator_controls_0.control_1').Time, sysvector('actuator_controls_0.control_1').Data);
+plot(sysvector('actuator_controls_0.control_2').Time, sysvector('actuator_controls_0.control_2').Data);
+plot(sysvector('actuator_controls_0.control_3').Time, sysvector('actuator_controls_0.control_3').Data);
+plot(sysvector('actuator_controls_0.control_4').Time, sysvector('actuator_controls_0.control_4').Data);
+legend('u_{ail}','u_{elev}', 'u_{red}', 'u_{throt}', 'u_{flaps}');
+ylabel('Act. outputs []')
 
-control(4) = subplot(4,1,4);
+axeshandle(end+1) = subplot_tight(nrSubplotSections,1,[9],[plotmargins.vert plotmargins.horiz]);
 % airspeed plots
 hold on;
 plot(sysvector('airspeed_0.true_airspeed_m_s').Time, sysvector('airspeed_0.true_airspeed_m_s').Data);
 plot(sysvector('airspeed_0.indicated_airspeed_m_s').Time, sysvector('airspeed_0.indicated_airspeed_m_s').Data);
 plot(sysvector('tecs_status_0.airspeedSp').Time, sysvector('tecs_status_0.airspeedSp').Data);
 % TODO add here v ref nom, v ref min, v ref max
 legend('v_{TAS} [m/s]','v_{IAS} [m/s]', 'v_{IAS} ref[m/s]');
-ylabel('Airspeed [m/s]')
-hold off;
-
-linkaxes(control(:),'x');
-set(control(:),'XGrid','on','YGrid','on','ZGrid','on');
+ylabel('Airsp. [m/s]')
 
-
-fig17 = figure(17);
```

Comment:
Can you remove the 17 from the PX4DisplayLogData (Line 43) as this figure will not be generated anymore.

Reply:
Done.

Reply:
The reason for the fixed number is that we can align the axes for the specific plots. Maybe we can get rid of them but I will test that tomorrow.

Reply:
Ok. In the old plot scripts we solved that without fixed plot numbers, which worked fine. I think it is somewhat easier.